### PR TITLE
feat: add ability to install docker-compose binary system-wide

### LINKED ISSUE #1974: Ask user to install compose binary to the system rather than adding to path
### Is your enhancement related to a problem? Please describe

When pressing the "compose" button, I'd like to be given the opportunity for the file to be copied over to `/usr/local/bin/`  rather than installed locally to `.local`

### Describe the solution you'd like

Option to install system-wide.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
feat: add ability to install docker-compose binary system-wide

### What does this PR do?

* Adds a prompt after installation to local storage directory, if we
  want to install docker-compose system wide.
* Added tests

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/6422176/80402f15-0402-42d4-af0c-3773e0d0bd17

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/1974

It does NOT close
https://github.com/containers/podman-desktop/issues/2709 though as the
issue with the "add to $PATH wrapper" still exists

### How to test this PR?

1. Remove your docker-compose binary from
   ~/.local/share/containers/podman-desktop
2. Test the functionality by re-installing compose by clicking the
   button
3. Type `whereis docker-compose` to confirm it has been installed
   system-wide

### CODE REVIEW

File: extensions/compose/src/cli-run.spec.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  let fakeOs: OS;
+  const cliRun = new CliRun(fakeContext, fakeOs);
+
+  // Run installBinaryToSystem with a non-binary file
+  try {
+    await cliRun.installBinaryToSystem('test', 'tmpBinary');
+    // Expect that showErrorMessage is called
+    expect(extensionApi.window.showErrorMessage).toHaveBeenCalled();
+  } catch (err) {
+    expect(err).to.be.a('Error');
```
Comment (jeffmaury) [inline extensions/compose/src/cli-run.spec.ts]: Why do we have those 2 expect ?

Comment (cdrage) [reply extensions/compose/src/cli-run.spec.ts]: To make sure that the message shows it as an actual 'Error'. Sometimes testing gives a false positive when it cant display the message correctly, so I always check the error output.

Comment (benoitf) [reply extensions/compose/src/cli-run.spec.ts]: for tests I think we don't use try/catch approach but use instead   expect (..call the method.).rejects.toThrow('error messages)

Comment (cdrage) [reply extensions/compose/src/cli-run.spec.ts]: Redid the test! Thanks for the input

Comment (benoitf) [reply extensions/compose/src/cli-run.spec.ts]: you can also use toThrow('error message') to verify which error message is thrown (if it's the expected one and not another error)

**Review by benoitf — APPROVED** [review]
tested and it worked fine

I was just wondering about 'existing Podman Desktop users'
if they launch Podman Desktop, they won't be notified if they want to install or not docker-compose to their system PATH
should we do something for them ?

**Review by lstocchi — APPROVED** [review]
LGTM

**Comment by cdrage** [comment]
> tested and it worked fine
> 
> I was just wondering about 'existing Podman Desktop users' if they launch Podman Desktop, they won't be notified if they want to install or not docker-compose to their system PATH should we do something for them ?

I don't think so, as it may introduce an edge-case of the prompt forever appearing.

We should introduce instructions on how to install via documentation changes instead.

**Pull request merged**